### DISCUSSION TITLE
How to close the auto-suggest list after pressing ESC, and keep in insert mode?

### DISCUSSION THREAD

**Discussion by hlf20010508** [neoclide/coc.nvim#4028] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**
I think it's not a problem, but I just don't know how to do it.
I'm used to press ESC to close the suggest list, but it will exit the insert mode too.
Is there any way to solve this?

**Describe the solution you'd like**
Press the ESC, and it should close the auto-suggest list, and keep in the insert mode.

**Describe alternatives you've considered**
I've tried this:
```
inoremap <silent><expr> <ESC> coc#pum#visible() ? coc#pum#stop() : "\<ESC>"
```
It can somehow do what I want, but it'll wait for about half a second to let the list close.
And I notice that in this way, after I press ESC, it exit insert mode, and after half a second the list closes and it turn back to insert mode. It's wierd but it does do what I want.
But is there any better way to do so?

**Comment by fannheyward** [comment]
I'm using this `inoremap <silent><expr> <C-e> coc#pum#visible() ? coc#pum#cancel() : "\<End>"`. 

Maybe `coc#pum#cancel()` is what you want.

**Comment by hlf20010508** [comment]
> I'm using this `inoremap <silent><expr> <C-e> coc#pum#visible() ? coc#pum#cancel() : "\<End>"`.
> 
> Maybe `coc#pum#cancel()` is what you want.

Thanks for your reply.
I've already tested `coc#pum#cancel()` and it did the same with `coc#pum#stop()`.
I think the problem is that ESC is not override by map, so it will do a normal ESC and then do `coc#pum#stop()` or `coc#pum#cancel()`. If I change ESC to `<C-l>` or other keys, it works well.

**Comment by chemzqm** [comment]
`:h coc#pum#stop()`

**Comment by chemzqm** [comment, marked as answer]
Your `<esc>` could be remapped, checkout `:verbose imap <esc>` or your need nowait `:h map-nowait`
